### ISSUE TITLE
problem installing sentencepiece from source

### ISSUE THREAD

**Issue by winobes** [google/sentencepiece#411]
Hello,

I encounter the following error installing sentencepiece from source:

```
$ pip install sentencepiece --no-binary :all:
Collecting sentencepiece
  Using cached https://files.pythonhosted.org/packages/1b/87/c3c2fa8cbec61fffe031ca9f0da512747520bec9be7f886f748457daac31/sentencepiece-0.1.83.tar.gz
    ERROR: Command errored out with exit status 1:
     command: ~/tmp-env/bin/python3.7 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/cp/kh9797c57j938rxn3wj9kscm0000gp/T/pip-install-04w9v7zw/sentencepiece/setup.py'"'"'; __file__='"'"'/private/var/folders/cp/kh9797c57j938rxn3wj9kscm0000gp/T/pip-install-04w9v7zw/sentencepiece/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/cp/kh9797c57j938rxn3wj9kscm0000gp/T/pip-install-04w9v7zw/sentencepiece/pip-egg-info
         cwd: /private/var/folders/cp/kh9797c57j938rxn3wj9kscm0000gp/T/pip-install-04w9v7zw/sentencepiece/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/cp/kh9797c57j938rxn3wj9kscm0000gp/T/pip-install-04w9v7zw/sentencepiece/setup.py", line 29, in <module>
        with codecs.open(os.path.join('..', 'VERSION'), 'r', 'utf-8') as f:
      File "~/tmp-env/bin/../lib/python3.7/codecs.py", line 904, in open
        file = builtins.open(filename, mode, buffering)
    FileNotFoundError: [Errno 2] No such file or directory: '../VERSION'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

It seems that the pypi source distribution is missing `VERSION`

unfortunately, i need to install it from source because my distribution isn't compatible with manylinux1

**Comment by n8henrie** [comment]
https://github.com/google/sentencepiece/releases -- a wheel for 3.8 exists but not yet pushed to pypi. I was able to work around with: `pip install https://github.com/google/sentencepiece/releases/download/v0.1.84/sentencepiece-0.1.84-cp38-cp38-manylinux1_x86_64.whl`

**Comment by taku910** [comment]
Fixed the issue of VERSION file in v0.1.85

**Comment by n8henrie** [comment]
This issue persists on MacOS because [MacOS did not have a 3.8 wheel released](https://github.com/google/sentencepiece/releases/tag/v0.1.85), and therefore v0.1.83 is still attempted with a pip install. Can we get a Mac wheel? (I see one for tf-sentencepiece.) Should I start a new issue for this?

**Comment by GDBSD** [comment]
I can confirm that this is a MacOS issue. Fix  is simple albeit a bit of a hack. I created an Anaconda virtual environment with Python 3.6 and the Transformers install was successful.

**Comment by adamhadani** [comment]
+1 - would be great to solve this for Python 3.8 on OSX, seems like we're just missing a wheel file for that distribution

**Comment by vijetadeshpande** [comment]
Is there any update on this? I tried to 

pip install https://github.com/google/sentencepiece/releases/download/v0.1.85/sentencepiece-0.1.85-cp27-cp27m-macosx_10_6_x86_64.whl

but did not work for python 3.8 on OSX.

**Comment by vijetadeshpande** [comment]
I was able to install it with following:

pip install https://github.com/google/sentencepiece/releases/download/v0.1.85/tf_sentencepiece-0.1.85-py2.py3-none-macosx_10_10_x86_64.whl

**Comment by n8henrie** [comment]
Correct, a cpython 2.7 wheel (cp27) is not likely to work for python 3. Glad you got it sorted out!

**Comment by physicsrob** [comment]
I'm also blocked by this issue.  Using python 3.8 on mac, and can't seem to get sentencepiece working.

**Comment by pmanresa** [comment]
I managed to finally get sentencepiece and transformers installation working on my mac. 

It worked when changing to python 3.6.10. But based on the comments above, I guess it would work for any 3.6.X.

**Comment by Kruszylo** [comment]
**MacOS: 10.14.6
Python 3.8.1**
 Found a way how to install `flair`. The problem was with `sentencepiece`, currently there is no package for `python 3.8` for Mac. Workaround is: 

- download source code `tar.gz` ([v0.1.85](https://github.com/google/sentencepiece/releases/tag/v0.1.85))
- unpackage and install:
```
cd /path/to/surce/code/
tar xopf sentencepiece-0.1.85.tar
cd sentencepiece-0.1.85/python/
pip install .
```
than:
```
pip install transformers==2.4.1
pip install flair
```
